From 36304436b9d8a9e8a299a61f875c92bbe2e7a57b Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Tue, 27 Nov 2012 12:20:55 -0800
Subject: build: Conditionally include iago Image creation Makefiles

Iago is the next-generation Diskinstaller for Legacy BIOS and EFI-
based Intel devices. It is a replacement for the existing
Diskinstaller program.

This file contains directives which cannot go in an Android.mk
and needs to be after they all have been included, similar to the
old Diskinstaller ramdisk directives.

Additionally include the Iago provision.mk file, which needs
to be included after the rules for the target-files-package.
These Makefile directives need to live in the same repo as
the rest of the Iago code and are hence not included in a
vendor/*/build/tasks directory.

Bug: ADEV-129
Category: Device Enablement
Domain: AOSP-build, INSTALLER-iago
Severity: High
Origin: Internal
Upstream-candidate: Yes, but only when we are ready to upstream
                    the bootable/iago project

Change-Id: I4c07f13a857f19052e15c562805f8f64452e4c3d
Signed-off-by: Andrew P. Boie <andrew.p.boie@intel.com>

Conflicts:
	core/Makefile
---
 core/Makefile      |    7 +++++++
 core/cleanbuild.mk |    4 +++-
 2 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 02b015a..056e6d0 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1105,6 +1105,9 @@ endif # BOARD_VENDORIMAGE_FILE_SYSTEM_TYPE
 ifeq ($(TARGET_USE_DISKINSTALLER),true)
 include bootable/diskinstaller/config.mk
 endif
+ifeq ($(TARGET_USE_IAGO),true)
+include bootable/iago/image.mk
+endif
 
 # -----------------------------------------------------------------
 # host tools needed to build dist and OTA packages
@@ -1665,6 +1668,10 @@ include $(sort $(wildcard $(BUILD_SYSTEM)/tasks/*.mk))
 $(call import-mixin-file,buildtask.mk)
 endif
 
+ifeq ($(TARGET_USE_IAGO),true)
+include bootable/iago/provision.mk
+endif
+
 # -----------------------------------------------------------------
 # Create SDK repository packages. Must be done after tasks/* since
 # we need the addon rules defined.
diff --git a/core/cleanbuild.mk b/core/cleanbuild.mk
index fa67332..7103140 100644
--- a/core/cleanbuild.mk
+++ b/core/cleanbuild.mk
@@ -179,6 +179,7 @@ installclean_files := \
 	$(HOST_OUT)/obj/NOTICE_FILES \
 	$(HOST_OUT)/sdk \
 	$(PRODUCT_OUT)/*.img \
+	$(PRODUCT_OUT)/*.iso \
 	$(PRODUCT_OUT)/*.txt \
 	$(PRODUCT_OUT)/*.xlb \
 	$(PRODUCT_OUT)/*.zip \
@@ -196,7 +197,8 @@ installclean_files := \
 	$(PRODUCT_OUT)/obj/EXECUTABLES/adbd_intermediates \
 	$(PRODUCT_OUT)/obj/EXECUTABLES/init_intermediates \
 	$(PRODUCT_OUT)/obj/ETC/mac_permissions.xml_intermediates \
-	$(PRODUCT_OUT)/obj/ETC/sepolicy_intermediates
+	$(PRODUCT_OUT)/obj/ETC/sepolicy_intermediates \
+	$(PRODUCT_OUT)/iago
 
 # The files/dirs to delete during a dataclean, which removes any files
 # in the staging and emulator data partitions.
-- 
1.7.4.1

